src/sentry/api/endpoints/organization_missing_org_members.py

```diff
@@ -74,40 +66,54 @@ def _get_missing_organization_members(
+    org_id = organization.id
+    domain_query = ""
     if shared_domain:
-        nonmember_authors = nonmember_authors.filter(email__endswith=shared_domain)
+        domain_query = f"AND sentry_commitauthor.email::text LIKE '%%{shared_domain}'"
     else:
-        for filtered_email in FILTERED_EMAIL_DOMAINS:
-            nonmember_authors = nonmember_authors.exclude(email__endswith=filtered_email)
-
-    for filtered_character in FILTERED_CHARACTERS:
-        nonmember_authors = nonmember_authors.exclude(email__icontains=filtered_character)
-
-    org_repos = Repository.objects.filter(
-        provider="integrations:" + provider,
-        organization_id=organization.id,
-        integration_id__in=integration_ids,
-    ).values_list("id", flat=True)
-
-    recent_commits = Commit.objects.filter(
-        repository_id__in=org_repos, date_added__gte=timezone.now() - timedelta(days=30)
-    ).values_list("id", flat=True)
-
-    return (
-        nonmember_authors.filter(commit__id__in=recent_commits)
-        .annotate(Count("commit"))
-        .order_by("-commit__count")
+        for filtered_email_domain in FILTERED_EMAIL_DOMAINS:
+            domain_query += (
+                f"AND sentry_commitauthor.email::text NOT LIKE '%%{filtered_email_domain}' "
+            )
+
+    date_added = (timezone.now() - timedelta(days=30)).strftime("%Y-%m-%d, %H:%M:%S")
+
+    query = f"""
+        SELECT sentry_commitauthor.id, sentry_commitauthor.organization_id, sentry_commitauthor.name, sentry_commitauthor.email, sentry_commitauthor.external_id, COUNT(sentry_commit.id) AS commit__count FROM sentry_commitauthor
+        INNER JOIN (
+            select * from sentry_commit
+            WHERE sentry_commit.organization_id = {org_id}
+            AND date_added >= '{date_added}'
+            order by date_added desc limit 1000
+        ) as sentry_commit ON sentry_commitauthor.id = sentry_commit.author_id
+
+        WHERE sentry_commit.repository_id IN
+        (
+            select id
+            from sentry_repository
+            where provider = 'integrations:{provider}'
+            and organization_id = {org_id}
+            and integration_id in ({", ".join([str(id) for id in integration_ids])})
+            )
+        AND sentry_commit.author_id IN
+            (select id from sentry_commitauthor
+                WHERE sentry_commitauthor.organization_id = {org_id}
+                AND NOT (
+                    (sentry_commitauthor.email IN (select concat(email, user_email) from sentry_organizationmember where organization_id = {org_id} and (email is not null or user_email is not null)
+                )
+        OR sentry_commitauthor.external_id IS NULL))
+        {domain_query}
+        AND NOT (UPPER(sentry_commitauthor.email::text) LIKE UPPER('%%+%%'))
+        )
+
+        GROUP BY sentry_commitauthor.id ORDER BY commit__count DESC limit 50
+    """.format(
+        org_id, date_added, provider, integration_ids, domain_query
     )
```

Comment:
You shouldn't format this directly, instead parameterise the query like https://docs.djangoproject.com/en/4.2/topics/db/sql/#passing-parameters-into-raw